common/stats/stats.go

```diff
@@ -425,28 +425,50 @@ type metricLatency struct {
 	metrics.Histogram
 	start     time.Time
 	precision time.Duration
+	mutex     sync.RWMutex
+}
+
+func (l *metricLatency) Time() Latency {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	l.start = Time.Now()
+	return l
+}
+
+func (l *metricLatency) Stop() {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	l.Update(Time.Since(l.start).Nanoseconds())
 }
-type nilLatency struct{}
 
-func (l *metricLatency) Time() Latency { l.start = Time.Now(); return l }
-func (l *metricLatency) Stop()         { l.Update(Time.Since(l.start).Nanoseconds()) }
 func (l *metricLatency) Capture() Latency {
-	return &metricLatency{l.Histogram.Snapshot(), l.start, l.precision}
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	return &metricLatency{l.Histogram.Snapshot(), l.start, l.precision, sync.RWMutex{}}
 }
+
 func (l *metricLatency) GetPrecision() time.Duration {
+	l.mutex.Lock()
```

Comment:
Can use Rlock/RUnlock GetPrecision. Otherwise LGTM

Reply:
yeah, that works. Updated.